makefile

```diff
@@ -1,8 +1,17 @@
-VERSION ?= 5.2.0
-BRANCH ?= develop
+VERSION  := 5.2.0
+BRANCH   := develop
+ARTIFACT := prodbin-$(VERSION)-$(BRANCH).tar.gz
 
-.PHONY: all
-all: tar
+.PHONY: all clean build javascript
 
-tar:
-	tar cfz prodbin-$(VERSION)-$(BRANCH).tar.gz Products bin
\ No newline at end of file
+all: build
+
+build: javascript
+	tar cvfz $(ARTIFACT) Products bin
+
+javascript:
+	make -f javascript.mk build
+
+clean:
+	rm -f $(ARTIFACT)
+	-make -f javascript.mk clean
```

Comment:
I'm guessing that if you try to clean and the files to be cleaned are absent it will exit non-zero, so the dash says to ignore that.